vcd/provider_test.go

```diff
@@ -36,28 +36,13 @@ func TestProvider_impl(t *testing.T) {
 }
 
 // When this function is called, the initialization in config_test.go has already happened.
-// Therefore, we can safely require that testConfig fields have been filled.
-func testAccPreCheck(t *testing.T) {
-	if testConfig.Provider.User == "" && testConfig.Provider.Token == "" {
-		t.Fatal("provider.user or provider.token must be set for acceptance tests")
-	}
-	if testConfig.Provider.Password == "" && testConfig.Provider.Token == "" {
-		t.Fatal("provider.password or provider.token must be set for acceptance tests")
-	}
-	if testConfig.Provider.SysOrg == "" {
-		t.Fatal("provider.sysOrg must be set for acceptance tests")
-	}
-	if testConfig.VCD.Org == "" {
-		t.Fatal("vcd.org must be set for acceptance tests")
-	}
-	if testConfig.Provider.Url == "" {
-		t.Fatal("provider.Url must be set for acceptance tests")
-	}
-	if testConfig.Networking.EdgeGateway == "" {
-		t.Fatal("networking.edgeGateway must be set for acceptance tests")
-	}
-	if testConfig.VCD.Vdc == "" {
-		t.Fatal("vcd.vdc must be set for acceptance tests")
+// Therefore, we can safely require that testConfig fields used in test params have been filled.
+// This function called moved from resource.Test.PreCheck before templateFill function call as  generates binary
```

Comment:
Could you rephrase - this one is hard to follow - I think it has some typos in here.

Reply:
done